tests/utils/CaretakerTest.cpp

```diff
@@ -0,0 +1,27 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include "doctest.h"
+#include "Caretaker.h"
+#include "Memento.h"
```

Comment:
also fix the includes here to `#include "Memento.h"` should be `#include "utils/Memento.h"` and `#include "Caretaker.h"` should be `#include "utils/Caretaker.h"`